packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -421,67 +443,48 @@ contract Deploy is Deployer {
-        // removed. If fault proofs are not enabled, the DisputeGameFactory proxy will be unused.
-        deployERC1967Proxy("DisputeGameFactoryProxy");
-        deployERC1967Proxy("DelayedWETHProxy");
-        deployERC1967Proxy("PermissionedDelayedWETHProxy");
-        deployERC1967Proxy("AnchorStateRegistryProxy");
-
-        deployAnchorStateRegistry();
-
-        // Deploy and setup the legacy (pre-faultproofs) contracts
-        if (!cfg.useFaultProofs()) {
-            deployERC1967Proxy("L2OutputOracleProxy");
-        }
-
-        initializeOpChain();
+        address superchainConfigProxy = mustGetAddress("SuperchainConfigProxy");
+        OPContractsManager opcm = OPContractsManager(mustGetAddress("OPContractsManagerProxy"));
+
+        OPContractsManager.DeployInput memory deployInput = getDeployInput();
+        OPContractsManager.DeployOutput memory deployOutput = opcm.deploy(deployInput);
+
+        // Save all deploy outputs from the OPCM, in the order they are declared in the DeployOutput struct
+        save("ProxyAdmin", address(deployOutput.opChainProxyAdmin));
+        save("AddressManager", address(deployOutput.addressManager));
+        save("L1ERC721BridgeProxy", address(deployOutput.l1ERC721BridgeProxy));
+        save("SystemConfigProxy", address(deployOutput.systemConfigProxy));
+        save("OptimismMintableERC20FactoryProxy", address(deployOutput.optimismMintableERC20FactoryProxy));
+        save("L1StandardBridgeProxy", address(deployOutput.l1StandardBridgeProxy));
+        save("L1CrossDomainMessengerProxy", address(deployOutput.l1CrossDomainMessengerProxy));
+
+        // Fault Proof contracts
+        save("DisputeGameFactoryProxy", address(deployOutput.disputeGameFactoryProxy));
+        save("PermissionedDelayedWETHProxy", address(deployOutput.delayedWETHPermissionedGameProxy));
+        save("AnchorStateRegistryProxy", address(deployOutput.anchorStateRegistryProxy));
+        save("AnchorStateRegistry", address(deployOutput.anchorStateRegistryImpl));
+        save("PermissionedDisputeGame", address(deployOutput.permissionedDisputeGame));
+        save("OptimismPortalProxy", address(deployOutput.optimismPortalProxy));
+
+        // Deploy and setup the PermissionlessDelayedWeth not provided by the OPCM
+        //   TODO: When the OPCM also deploys the PermissionlessGame, we will be able to simply call
+        //   save("PermissionlessDelayedWETHProxy", address(deployOutput.delayedWETHPermissionlessGameProxy));
+        address delayedWETHImpl = mustGetAddress("DelayedWETH");
+        address delayedWETHPermissionlessGameProxy = deployERC1967ProxyWithOwner("DelayedWETHProxy", msg.sender);
+        vm.broadcast(msg.sender);
+        IProxy(payable(delayedWETHPermissionlessGameProxy)).upgradeToAndCall({
+            _implementation: delayedWETHImpl,
+            _data: abi.encodeCall(IDelayedWETH.initialize, (msg.sender, ISuperchainConfig(superchainConfigProxy)))
+        });
```

Comment:
We could read the permissionless game implementation address from `gameImpls` and revert here if it's not the zero address, to force this TODO to be addresses once OPCM deployes the permissionless game

Reply:
I like it, will do.

Reply:
3287bdcdd1ab